substrate/client/rpc-spec-v2/src/chain_head/chain_head.rs

```diff
@@ -180,12 +192,20 @@ where
 		let subscriptions = self.subscriptions.clone();
 		let backend = self.backend.clone();
 		let client = self.client.clone();
+		let rpc_connections = self.rpc_connections.clone();
 
 		let fut = async move {
+			// Ensure the current connection ID has enough space to accept a new subscription.
+			let connection_id = pending.connection_id();
+
+			let Some(reserved_token) = rpc_connections.reserve_token(connection_id) else {
```

Comment:
ok, you must return a JSON-RPC error if the limit is exceeded and that's why you look up the state in the hashmap before "subscription accept is called"?

Too bad to lock the mutex twice for that...  but no way around that I guess

Reply:
Can we merge `subscriptions` and `connections`? The subscription ID is stored in both and would be nice to merge them is possible